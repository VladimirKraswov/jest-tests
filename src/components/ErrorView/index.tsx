import React, {FC} from 'react';
import { useSelector } from 'react-redux';

import {IState, selectors} from '../../store';

interface IErrorViewProps {
  style: any
}

export const  ErrorView: FC<IErrorViewProps> = ({ style }) => {
  const error = useSelector<IState, string>(selectors.getError);

  if (!error) {
    return null;
  }

  return (
    <p style={style}>{error}</p>
  );
}