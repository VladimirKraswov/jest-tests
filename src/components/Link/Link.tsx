import {FC, useState} from 'react';

const STATUS = {
  HOVERED: 'hovered',
  NORMAL: 'normal',
};

interface ILinkProps {
  page: string
  children: React.ReactElement | string
}

export const  Link: FC<ILinkProps> = ({ page, children }) => {
  const [status, setStatus] = useState(STATUS.NORMAL);

  const onMouseEnter = () => {
    setStatus(STATUS.HOVERED);
  };

  const onMouseLeave = () => {
    setStatus(STATUS.NORMAL);
  };

  return (
    <a
      className={status}
      href={page || '#'}
      onMouseEnter={onMouseEnter}
      onMouseLeave={onMouseLeave}
    >
      {children}
    </a>
  );
}