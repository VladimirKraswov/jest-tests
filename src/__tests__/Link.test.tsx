// Тестирование при помощи снимков
import React from 'react';
import renderer from 'react-test-renderer';

import { Link } from '../components/Link';

it('Изменяет состояние при наведении курсора', () => {
  const component = renderer.create(
    <Link page="login">Login</Link>,
  );
  // Отрисуем компонент
  let tree: any = component.toJSON();
  expect(tree).toMatchSnapshot();

  // Имитируем наведение мышкой
  renderer.act(() => {
    tree.props.onMouseEnter();
  });
  // Перерисовываем компонент
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // Имитируем уход мышки
  renderer.act(() => {
    tree?.props.onMouseLeave();
  });
  // Перерисовываем компонент
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});