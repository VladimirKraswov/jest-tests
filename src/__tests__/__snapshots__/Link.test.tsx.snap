// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Изменяет состояние при наведении курсора 1`] = `
<a
  className="normal"
  href="login"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Login
</a>
`;

exports[`Изменяет состояние при наведении курсора 2`] = `
<a
  className="hovered"
  href="login"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Login
</a>
`;

exports[`Изменяет состояние при наведении курсора 3`] = `
<a
  className="normal"
  href="login"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Login
</a>
`;
